linen_examples/seq2seq/train.py

```diff
@@ -130,12 +130,13 @@ def encode_onehot(batch_inputs, max_len):
 
   def encode_str(s):
     tokens = CTABLE.encode(s)
-    if len(tokens) > max_len:
+    unpadded_len = len(tokens)
+    if unpadded_len > max_len:
       raise ValueError(f'Sequence too long ({len(tokens)}>{max_len}): \'{s}\'')
     tokens = np.pad(tokens, [(0, max_len-len(tokens))], mode='constant')
-    return onehot(tokens, CTABLE.vocab_size)
+    return onehot(tokens, CTABLE.vocab_size), unpadded_len
 
-  return np.array([encode_str(inp) for inp in batch_inputs])
+  return np.array([encode_str(inp) for inp in batch_inputs], dtype=object)
```

Comment:
What is the point of wrapping this in a `np.array(..., dtype=object)`?  Don't the below uses work just as well if this were just a list?

Reply:
If I don't add this then I get the following warning:

```
train.py:139: VisibleDeprecationWarning: Creating an ndarray from ragged nested sequences (which is a list-or-tuple of lists-or-tuples-or ndarrays with different lengths or shapes) is deprecated. If you meant to do this, you must specify 'dtype=object' when creating the ndarray
  return np.array([encode_str(inp) for inp in batch_inputs])
```

Reply:
but why wrap this in an ndarray at all??  why not just keep it as a list?

Reply:
Ahh yes that actually works! Nice, changed it.